extensions/scarb-snforge-test-collector/src/main.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use std::str::FromStr;
+
+use scarb_metadata::MetadataCommand;
+
+use crate::compilation::compile_tests;
+use crate::crate_collection::collect_test_compilation_targets;
+use crate::metadata::{
+    corelib_for_package, dependencies_for_package, id_for_package, paths_for_package,
+};
+
+mod compilation;
+mod crate_collection;
+mod metadata;
+
+/// Starknet Foundry private extension for compiling test artifacts.
+/// Users should not call it directly.
+#[derive(Parser)]
+#[command(version)]
+struct Args {
+    #[arg(long)]
+    output_path: Utf8PathBuf,
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+    let scarb_metadata = MetadataCommand::new().inherit_stderr().exec()?;
+
+    let manifest_path = Utf8PathBuf::from_str(&std::env::var("SCARB_MANIFEST_PATH")?)?;
+    let package_id = id_for_package(&scarb_metadata, &manifest_path)?;
+
+    let (package_path, package_source_dir_path, package_name) =
```

Comment:
I feel like this flattening of project abstraction is actually more confusing?
You could pass `PackageMetadata` (extracted with packages_filter) downwards, and extract info you need, where you actually need it.